imap/user.c

```diff
@@ -328,6 +328,12 @@ EXPORTED int user_deletedata(const mbentry_t *mbentry, int wipe_user)
     search_deluser(mbentry);
 
 #ifdef WITH_DAV
+    /* delete the entire DAV DB */
+    struct buf fname = BUF_INITIALIZER;
+    dav_getpath_byuserid(&fname, userid);
+    unlink(buf_cstring(&fname));
+    buf_free(&fname);
+
```

Comment:
This does seem to fix the FastMail.rename_deepfolder... failures, but it's odd that it's necessary.  To my reading "dav" is one of the `user_file_suffixes`, so it should have been added to `paths`, and then deleted by the loop over `paths` a few lines below??

Reply:
Yeah, you're right.  This was my first attempt at a fix which didn't work.  The real fix was actually NOT creating the DAV DB when the user INBOX doesn't yet exist.  Reverting this commit.